lib/vector-core/src/stream/batcher.rs

```diff
@@ -0,0 +1,540 @@
+use crate::partition::Partitioner;
+use crate::time::Timer;
+use crate::ByteSizeOf;
+use futures::stream::Stream;
+use pin_project::pin_project;
+use std::collections::HashMap;
+use std::hash::BuildHasherDefault;
+use std::mem;
+use std::num::NonZeroUsize;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+use tokio::time::{interval, Interval, MissedTickBehavior};
+use twox_hash::XxHash64;
+
+/// A [`Timer`] for [`Batch`]
+///
+/// A `Batch` implementation must be rigged to flush every so often on a
+/// timer. Absent any particular concerns this timer will serve most
+/// purposes. It stores an internal notion of when it was started and how "wide"
+/// away from that start time can drift before the timer has elapsed.
+pub struct BatcherTimer {
+    interval: Interval,
+}
+
+impl BatcherTimer {
+    #[allow(dead_code)]
+    pub fn new(period: Duration) -> Self {
+        let mut interval = interval(period);
+        interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
```

Comment:
Would `Delay` behavior be preferable to `Skip` here? They are similar, except that the _next_ interval starts from whenever you call `tick` instead of at a regular interval.

Reply:
I think it would, yes.

Reply:
Resolved in 0071ae498